snow/engine/snowman/bootstrap/bootstrapper.go

```diff
@@ -603,25 +617,36 @@ func (b *Bootstrapper) process(
 		height := blk.Height()
 		b.tipHeight = max(b.tipHeight, height)
 
-		if numPreviouslyFetched/statusUpdateFrequency != numFetched/statusUpdateFrequency {
+		// Check if it's time to log progress (both progress-based and time-based frequency)
+		now := time.Now()
+		shouldLog := numPreviouslyFetched/statusUpdateFrequency != numFetched/statusUpdateFrequency &&
+			now.Sub(b.lastProgressUpdateTime) >= minimumLogInterval
+
+		if shouldLog {
 			totalBlocksToFetch := b.tipHeight - b.startingHeight
-			eta := timer.EstimateETA(
-				b.startTime,
-				numFetched-b.initiallyFetched,         // Number of blocks we have fetched during this run
-				totalBlocksToFetch-b.initiallyFetched, // Number of blocks we expect to fetch during this run
+
+			eta, progressPercentage := b.etaTracker.AddSample(
```

Comment:
I wish the log rate didn't modify how the eta updates... I suppose it is just up to anyone modifying this to know that doubling the logging rate should also double the number of internally tracked samples in the eta tracker for the same eta volatility.

It might be possible to do some ewma instead to avoid this (like done [here](https://github.com/ava-labs/avalanchego/blob/master/utils/math/continuous_averager.go#L11))... but it isn't immediately obvious to me how best to do that.